File: practices/practice5.1/src/MegaData.cpp
#include "MegaData.hpp"
#include <algorithm>

MegaData::MegaData() {
    reset();
}

void MegaData::reset() {
    std::ranges::fill(smallArray, DEFAULT_VALUE);
    std::ranges::fill(bigArray, DEFAULT_VALUE);
}

double* MegaData::getBigArray() {
    return bigArray;
}

float* MegaData::getSmallArray() {
    return smallArray;
}

File: homeworks/homework5.1/src/main/java/org/example/MegaData.java
package org.example;

import java.util.Arrays;

public class MegaData {
    public static final int DATA_COUNT = 1024;
    private final float[] smallArray = new float[DATA_COUNT];
    private final double[] bigArray = new double[DATA_COUNT * DATA_COUNT];

    public MegaData() {
        reset();
    }

    public void reset() {
        Arrays.fill(smallArray, 42.0f);
        Arrays.fill(bigArray, 42.0);
    }

    public float[] getSmallArray() {
        return smallArray;
    }

    public double[] getBigArray() {
        return bigArray;
    }
}
